src/tir/schedule/primitive/compute_inline.cc

```diff
@@ -295,7 +295,8 @@ class BaseInliner : public StmtExprMutator {
    * \param stmt The statement in which to count undefined variables
    */
   static int GetNumUndefinedNonpointerVars(const Stmt& stmt) {
-    auto undefined_vars = UndefinedVars(stmt, {});
+    // Do not visit buffer shape/strides.
+    auto undefined_vars = UndefinedVars(/*stmt=*/stmt, /*defs=*/{}, /*visit_buffer=*/false);
```

Comment:
If I'm reading it correctly, it looks like this would also ignore variables that are used in the shape/stride of buffers that are declared internal to `stmt`, but the suppression of the undefined variable failures should only apply to buffers that are declared externally to `stmt`.

Instead of defining contexts in which variables should not be considered, could we instead define variables should be ignored by `UndefinedVars`?  If a variable's definition occurs outside the LCA of the producer/consumer, then it would be legal to use in the inlined statement.  The dynamic buffer shapes would then need to follow this more general condition, rather than being ignored entirely.

Reply:
As I'm thinking on it, this would also allow `compute_inline` to apply in cases that are currently forbidden, but are semantically valid.  In the example below, the block `"B"` couldn't be inlined, because it refers to the variable `value`.  However, `value`'s definition is accessible from both block `"B"` and block `"C"`, so it could be inlined without risk of an undefined variable.

```python
@T.prim_func
def func(A: T.Buffer(64, "int32"), C: T.Buffer(64, "int32")):
    B = T.alloc_buffer(64, "int32")

    value: T.int32 = 42

    for i in range(64):
        with T.block("B"):
            vi = T.axis.remap("S", [i])
            B[vi] = value

    for i in range(64):
        with T.block("C"):
            vi = T.axis.remap("S", [i])
            C[vi] = value
```

Reply:
@Lunderberg Thanks for your suggestion!
I agree with you that collecting variables outside LCA producer/consumer is a better idea. I'll revise the code accordingly and ping you again once it's ready.

Reply:
Hi @Lunderberg sorry for being late, would you mind helping review #14451 instead? The `GetNumUndefinedNonpointerVars ` check doesn't seem to work on nested blocks (in dynamic workloads we always blockize the dynamic loop, which would emit nested blocks). So I prefer moving on with the solution in #14451 which uses a simpler check.